src/cc_catalog_airflow/dags/util/loader/sql.py

```diff
@@ -288,3 +289,96 @@ def _delete_malformed_row_in_file(tsv_file_name, line_number):
             if index + 1 != line_number:
                 write_obj.write(line)
 
+
+def _create_temp_sub_prov_table(
+        postgres_conn_id,
+        temp_table='temp_sub_prov_table'
+):
+    """
+    Drop the temporary table if it already exists
+    """
+    postgres = PostgresHook(postgres_conn_id=postgres_conn_id)
+    postgres.run(f'DROP TABLE IF EXISTS public.{temp_table};')
+
+    """
+    Create intermediary table for sub provider migration
+    """
+    postgres.run(
+        dedent(
+            f'''
+            CREATE TABLE public.{temp_table} (
+              {col.CREATOR_URL} character varying(2000),
+              {col.PROVIDER} character varying(80)
+            );
+            '''
+        )
+    )
+    postgres.run(
+        f'ALTER TABLE public.{temp_table} OWNER TO {DB_USER_NAME};'
+    )
+
+    """
+    Populate the intermediary table with the sub providers of interest
+    """
+    for sub_prov, user_id_set in prov.FLICKR_SUB_PROVIDERS.items():
+        for user_id in user_id_set:
+            creator_url = prov.FLICKR_PHOTO_URL_BASE + user_id
+            postgres.run(
+                dedent(
+                    f'''
+                    INSERT INTO public.{temp_table} (
+                      {col.CREATOR_URL} ,
+                      {col.PROVIDER}
+                    ) 
+                    VALUES (
+                      '{creator_url}' , 
+                      '{sub_prov}'
+                    );
+                    '''
+                )
+            )
+
+    return temp_table
+
+
+def update_sub_providers(
+        postgres_conn_id,
+        image_table=IMAGE_TABLE_NAME,
+        default_provider=prov.FLICKR_DEFAULT_PROVIDER
+):
+    """
+    Initially set all source values to the default provider value
+    """
+    postgres = PostgresHook(postgres_conn_id=postgres_conn_id)
+    postgres.run(
+        dedent(
+            f'''
+            UPDATE {image_table}
+            SET {col.SOURCE} = '{default_provider}'
+            WHERE {col.PROVIDER} = '{default_provider}';
+            '''
+        )
+    )
+
+    """
+    Update the source value to appropriate sub provider value for a given set
+    of users
+    """
+    temp_table = _create_temp_sub_prov_table(postgres_conn_id)
+    postgres.run(
+        dedent(
+            f'''
+            UPDATE {image_table}
+            SET {col.SOURCE} = public.{temp_table}.{col.PROVIDER}
+            FROM public.{temp_table}
+            WHERE 
+            {image_table}.{col.CREATOR_URL} = public.{temp_table}.{
+            col.CREATOR_URL};
```

Comment:
We will have to test this at scale to see whether we need an index to make this workable.  However, we could always add the index within this function, use it, then drop it (to avoid slowing down other things.  If the index is added concurrently, that wouldn't block too much.  I think it might be worth it, since we're looping through a number of creator URLs (and that number is expected to grow); we'd get to reuse the index.